pre-commit/check.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python
+
+from typing import List, Union, Dict, Set
+from pathlib import Path
+import argparse
+import os
+import sys
+
+# TODO add option to run rules in parallel
+# TODO add option to pass in a git diff to check or git hash
+
+
+def get_git_commit_files(commit_hash: str) -> List[Path]:
+    output = os.popen(f"git diff-tree --no-commit-id --name-only -r {commit_hash}").read()
+    lines = output.split("\n")
+    lines = [x for x in lines if x]
+    files = [Path(x).absolute() for x in lines]
+    return files
+
+
+# get a list of paths to dirty and staged files from git
+def get_git_status() -> List[Path]:
+    git_status = os.popen("git status --porcelain").read()
+    git_status_list = git_status.split("\n")
+    git_status_list = [x for x in git_status_list if x]
+    files = [Path(x.split()[1]).absolute() for x in git_status_list]
+    return files
+
+
+# gets last changed files
+def get_changed_files() -> List[Path]:
+    files = get_git_status()
+    if len(files) != 0:
+        return files
+    files = get_git_commit_files("HEAD")
+    return files
+
+
+# check if path a child of another path
+def is_child(path: Path, parent: Path) -> bool:
```

Comment:
I'm thinking this can be replaced by calling the `is_relative_to` function:
https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.is_relative_to

Reply:
It looks like we can't use that yet. `New in version 3.9.`